tests/network/vmi_istio.go

```diff
@@ -0,0 +1,244 @@
+			Masquerade: &v1.InterfaceMasquerade{},
+		},
+	}
+	interfaceWithExplicitPorts := v1.Interface{
+		Name: "default",
+		InterfaceBindingMethod: v1.InterfaceBindingMethod{
+			Masquerade: &v1.InterfaceMasquerade{},
+		},
+		Ports: []v1.Port{
+			{
+				Port: svcDeclaredTestPort,
+			},
+			{
+				Port: svcUndeclaredTestPort,
+			},
+		},
+	}
+
+	Describe("Virtual machine in a service mesh using masquerade interface with explicit port list", func() {
+		tests.BeforeAll(func() {
+			if !istioServiceMeshDeployed() {
+				Skip("Istio service mesh is required for service-mesh tests to run")
+			}
+
+			tests.BeforeTestCleanup()
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			vmiWithExplicitPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiExplicitPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&interfaceWithExplicitPorts,
+			)
+			vmiWithDefaultPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiDefaultPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&defaultInterface,
+			)
+			vmiWithExplicitPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithExplicitPorts.Name), vmiWithExplicitPorts.Namespace, vmiExplicitPortsAppSelector)
+			vmiWithDefaultPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithDefaultPorts.Name), vmiWithDefaultPorts.Namespace, vmiDefaultPortsAppSelector)
+
+			addLabelToNamespace(vmiWithExplicitPorts.Namespace, istioInjectNamespaceLabel, "enabled")
+			for _, svc := range []corev1.Service{*vmiWithExplicitPortsService, *vmiWithDefaultPortsService} {
+				_, err = virtClient.CoreV1().Services(svc.Namespace).Create(context.Background(), &svc, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+			}
+			for _, networkVMI := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
+				_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(networkVMI)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			// Wait for VMIs to become ready
+			vmiWithExplicitPorts = tests.WaitUntilVMIReady(vmiWithExplicitPorts, libnet.WithIPv6(console.LoginToCirros))
+			vmiWithDefaultPorts = tests.WaitUntilVMIReady(vmiWithDefaultPorts, libnet.WithIPv6(console.LoginToCirros))
+
+			for _, vmi := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
+				tests.StartHTTPServer(vmi, svcDeclaredTestPort)
+				tests.StartHTTPServer(vmi, svcUndeclaredTestPort)
+			}
+		})
+
+		table.DescribeTable("should be able to reach", func(destination string, port int) {
+			var srcVMI, dstVMI *v1.VirtualMachineInstance
+			switch destination {
+			case "vmiWithExplicitPorts":
+				srcVMI = vmiWithDefaultPorts
```

Comment:
To make this test lighter, I would use a pod as the src instead of a vm. Is there a good reason to use a vm?

Reply:
> Is there a good reason to use a vm?

My resoning is the following, I'm using both VMs in the 4 test cases, in 2 testcases, on acts as server and the other as a client, and in the other 2 test cases, it's the other way around. Since I need an associated service distinquished by a selector, I need 2 VMIs eventually.

I suppose I could serialize this so that for the first 2 cases I create 1 vmi  and for the other 2 cases I create another one. That would save 1 vmi worth of resources during the peak in exchange for time.